#165763690 Refactor checkAuth Helper Function

LordUche opened the pull request:
#### What does this PR do?
Refactors the `checkAuth` helper function
#### Description of Task to be completed?
- decouple the `checkAuth` from the redux store implementation
- dispatch `authSuccess` or `clearError` actions in the `componentDidMount` lifecycle hook in the `App` component
- write test
#### How should this be manually tested?
- check that the app still works from the [review app](https://vale-ah-frontend-staging-pr-13.herokuapp.com/)
- ensure all tests pass on `Travis CI`
#### Any background context you want to provide?
The `checkAuth` helper function should not be coupled with the redux store or dispatch any action.
#### What are the relevant pivotal tracker stories?
[#165763690](https://www.pivotaltracker.com/story/show/165763690)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

[Finishes #165763690]

LordUche committed:
chore(checkAuth): refactor checkAuth helper function

andela-pessien commented on src/App.jsx:
```diff
@@ -13,16 +15,14 @@ class App extends React.Component {
-    });
+  async componentDidMount() {
+    const user = await checkAuth();
+    if (user) store.dispatch(authSuccessAction({ user }));
```
Directly `dispatch`ing actions within a component is an antipattern; that is what `connect` is for.

LordUche replied:
The `App` component has the `Provider` as a child component so I can't connect to the child. That's why it was done this way. I will make `Provider` the parent of `App` in `index.jsx`  and remove it from this file. That way, I'll be able to connect to the store from here.

andela-pessien commented on src/utils/helpers.js:
```diff
@@ -20,17 +19,17 @@ export const normalizeErrors = error => {
       headers: { authorization: token },
     });
-    store.dispatch(action.authSuccessAction({ user: res.data.user }));
+    return res.data;
```
I think I was not clear enough initially - my point with this function was, why was this manually coupled to the store implementation when it could simply be an action dispatcher?

LordUche replied:
Yeah, I agree that's a better way to go about it 🙂

LordUche committed:
chore(checkAuth): refactor checkAuth

fix(helpers.test.jsx): remove checkAuth test

test(auth): refactor auth tests

fix(toast): fix toast message display issue

LordUche committed:
chore(toast): fix toast message issue

LordUche committed:
chore(auth): refactor auth test
